settings.lua

```diff
@@ -1630,40 +1625,29 @@ end
 --- Set the next value
 function DriverModeSetting:setNext()
 	if not self:isDisabled() then
-		local new = self:getNextMode()
-		self:setToIx(new)
+		for i=1,self.NUM_MODES do 
```

Comment:
Would something like this work? I really don't like reimplementing SettingList:setNext()...
```lua
repeat    
   SettingList.setNext(self)
until courseplay:getIsToolCombiValidForCpMode(self.vehicle, self:get())

```

Reply:
I see what you mean. This will probably be a problem then, as the value will get set to default(mode 5) everytime a none valid mode was tried. Probably would need logic to check if the mode swap is valid (4->5) or (4->6[not allowed] => 5) : 
`function DriverModeSetting:checkAndSetValidValue(new)
	if courseplay:getIsToolCombiValidForCpMode(self.vehicle,new) then 
		return SettingList.checkAndSetValidValue(self,new)
	else 
		return self.defaultMode
	end
end
`

Reply:
I see. How about this?
```lua
local targetMode = self.current
repeat
   targetModeIx = SettingList.checkAndSetValidValue(self, self.current + 1)
until courseplay:getIsToolCombiValidForCpMode(self.vehicle, self.values[targetModeIx])
self:setToIx(targetModeIx)
```